src/components/NavigationBar/NavigationBar.tsx

```diff
@@ -0,0 +1,335 @@
+import styled from "@emotion/styled";
+import React, { useState, FunctionComponent } from "react";
+import { Icon } from "react-feather";
+
+export enum NavigationItemType {
+  NavigationLink = "NavigationLink",
+  DropDownList = "DropDownList",
+  LabelButton = "LabelButton",
+  IconButton = "IconButton",
+}
+
+interface NavigationLink {
+  schema: NavigationItemType.NavigationLink;
+  id: string;
+  label: string;
+  className?: string;
+  path: string;
+  position: "left" | "center" | "right";
+}
+
+interface DropDownList extends Omit<NavigationLink, "schema" | "path"> {
+  schema: NavigationItemType.DropDownList;
+  dropdownItems: { id: string; label: string; path: string }[];
+}
+
+interface LabelButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.LabelButton;
+}
+
+interface IconButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.IconButton;
+  icon: Icon;
+}
+
+export type NavigationItem = NavigationLink | DropDownList | LabelButton | IconButton;
+
+export interface NavigationBarProps {
+  navigationItems: NavigationItem[];
+}
+
+const NavigationBarStyle = styled.nav`
+  button {
+    color: #6e787f;
+    font-size: 16px;
+    font-family: "Roboto";
+  }
+
+  a {
+    color: #6e787f;
+    display: block;
+    font-size: 16px;
+    font-family: "Roboto";
+  }
+
+  .navbar-toggler {
+    padding: 0.25rem 0.75rem;
+    background-color: transparent;
+    border: 1px solid transparent;
+    border-radius: 0.25rem;
+
+    &:hover,
+    &:focus {
+      text-decoration: none;
+      outline: none;
+    }
+
+    .icon-bar {
+      background-color: #454b50;
+      display: block;
+      width: 22px;
+      height: 2px;
+      border-radius: 1px;
+      transition: all 0.2s;
+    }
+
+    .icon-bar + .icon-bar {
+      margin-top: 10px;
+    }
+
+    .top-bar {
+      transform: rotate(45deg);
+      transform-origin: 10% 10%;
+    }
+
+    .bottom-bar {
+      transform: rotate(-45deg);
+      transform-origin: 10% 90%;
+    }
+
+    &.collapsed .top-bar {
+      transform: rotate(0);
+    }
+
+    &.collapsed .bottom-bar {
+      transform: rotate(0);
+    }
```

Comment:
```suggestion
    .icon-bar {
      background-color: #454b50;
      display: block;
      width: 22px;
      height: 2px;
      border-radius: 1px;
      transition: transform 0.2s ease-out;
      position: absolute
    }

    .top-bar {
      transform: translateY(0) rotate(45deg);
    }

    .bottom-bar {
      transform: translateY(0) rotate(-45deg);
    }

    &.collapsed .top-bar {
      transform: translateY(-4px) rotate(0);
    }

    &.collapsed .bottom-bar {
      transform: translateY(4px) rotate(0);
    }
```

agaration with `transform-origin` don't really look / feel nice, transition if can try to define the transition property. then if possible try to use as much tw classes rather than writing scoped styles (only if necessary)